components/camel-elytron/src/main/java/org/apache/camel/component/elytron/ElytronComponent.java

```diff
@@ -0,0 +1,163 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.elytron;
+
+import io.undertow.security.handlers.AuthenticationCallHandler;
+import io.undertow.security.handlers.AuthenticationConstraintHandler;
+import io.undertow.server.HttpHandler;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.undertow.HttpHandlerRegistrationInfo;
+import org.apache.camel.component.undertow.UndertowComponent;
+import org.apache.camel.component.undertow.UndertowEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.annotations.Component;
+import org.wildfly.elytron.web.undertow.server.ElytronContextAssociationHandler;
+import org.wildfly.elytron.web.undertow.server.ElytronRunAsHandler;
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.auth.server.MechanismConfiguration;
+import org.wildfly.security.auth.server.MechanismConfigurationSelector;
+import org.wildfly.security.auth.server.MechanismRealmConfiguration;
+import org.wildfly.security.auth.server.SecurityDomain;
+import org.wildfly.security.auth.server.http.HttpAuthenticationFactory;
+import org.wildfly.security.http.HttpAuthenticationException;
+import org.wildfly.security.http.HttpConstants;
+import org.wildfly.security.http.HttpServerAuthenticationMechanismFactory;
+import org.wildfly.security.http.util.FilterServerMechanismFactory;
+import org.wildfly.security.http.util.SecurityProviderServerMechanismFactory;
+
+import javax.net.ssl.SSLContext;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Provider;
+import java.util.Collections;
+
+/**
+ * Elytron component brings elytron security over came-undertow component.
+ *
+ * Component work either as producer and as consumer.
+ *
+ * User has to define its SecurityDomain.Builder which will be used for creation of security domain.
+ * MechanismName then allows to define mechanism, which will take care of authentication from security context.
+ * MechanusmName should be selected with regard to default securityRealm.
+ *
+ * Example: to use bearer_token, mechanism name has to be "BEARER_TOKEN" and realm has to be TokenSecurityRealm.
+ *
+ * @author JiriOndrusek
+ */
+@Metadata(label = "verifiers", enums = "parameters,connectivity")
+@Component("elytron")
+public class ElytronComponent extends UndertowComponent {
+
+    private static final WildFlyElytronProvider elytronProvider = new WildFlyElytronProvider();
```

Comment:
Hey @JiriOndrusek, it got to my attention that `WildFlyElytronProvider` is deprecated as I can see [here](https://github.com/wildfly-security/wildfly-elytron/blob/1.11.0.CR3/base/src/main/java/org/wildfly/security/WildFlyElytronProvider.java#L43), since I am not detailed about WildFlyElectron project, do you have an idea what is the alternative way to do it?

Reply:
Hi, I've noticed this a few hours ago as I was writing a test. I agree that we should use correct object, I'll look into it